pkg/operator/ceph/object/controller.go

```diff
@@ -319,3 +329,32 @@ func updateStatus(client client.Client, name types.NamespacedName, status string
 	}
 	logger.Debugf("object store %q status updated to %q", name, status)
 }
+
+func (r *ReconcileCephObjectStore) handleDelete(objectstore *cephv1.CephObjectStore) error {
+	//  Wait for all the object buckets to be cleaned up
+	err := r.waitForObjectBucketCleanup(objectstore)
+	if err != nil {
+		return errors.Wrap(err, "failed to cleanup buckets")
```

Comment:
I don't think this is the error here, we fail because there are OB still present